src/ray/id.cc

```diff
@@ -1,23 +1,38 @@
 #include "ray/id.h"
 
 #include <limits.h>
+
+#include <chrono>
+#include <mutex>
 #include <random>
 
 #include "ray/constants.h"
 #include "ray/status.h"
 
 namespace ray {
 
+std::mt19937 RandomlySeededMersenneTwister() {
```

Comment:
It might be better to inline this function, to avoid a copy ctor on return.

Reply:
Note that the copy will be optimized out due to https://en.cppreference.com/w/cpp/language/copy_elision, so I'll just leave it this way for now :)

Reply:
I actually know that, but it's good practice not to rely on compiler optimizations for that. Inlining the function will also get rid of the callq jump and the corresponding creation of the stack activation record, which _may_ also be optimized by the compiler. But why not do it explicitly, if that's what we want to happen?

Reply:
Please feel free to change it if you want to, but all modern C++ compilers do this optimization + in this case it is only called once per program run anyways (it is static) so it doesn't matter, and in general making too many things inline is also problematic since this leads to code blowup as I'm sure you know (which doesn't matter in this case of course since it is only called once). It's a tradeoff like always :)